### ISSUE TITLE
[a11y] TalkBack/Screenreader navigates to the previous screen when 'tapping' on PopupMenuItem

### ISSUE THREAD

**Issue by SexyHenk** [flutter/flutter#165182]
### Steps to reproduce

It seems like there is an issue with the combination of PopupMenuItem and TalkBack/screenreader. When TalkBack/Screenreader is enabled and you 'tap' on the PopupMenuItem, the handleTap is executed, which leads to navigation to the previous page. This does not happen without TalkBack/screenreader, which executes the OnTap within the PopupMenuItem.

### Expected results

While using TalkBack/Screenreader, tapping on the PopupMenuItem should execute the OnTap function.

### Actual results

While using TalkBack/Screenreader, tapping on the PopupMenuItem leads to the previous page.

### Code sample

<details open><summary>Code sample</summary>

```dart
  @override
  Widget build(BuildContext context) {
    return BlocConsumer<FilePickerCubit, FilePickerState>(
        bloc: _cubit,
        listener: _listen,
        builder: (BuildContext context, FilePickerState state) {
          if (state is FilePickerReadyState) {
            return PopupMenuItem<dynamic>(
              child: Semantics(
                label: widget.errorText,
                child: ActionTile(
                  title: _localization.fileSourceNewPhoto,
                  onTap: () {
                    _cubit.pickFromCamera(showTips: widget.showCameraTips);
                  },
                ),
              ),
            );
          }
        });
  }

```
In PopupMenuEntry:
```dart

@protected
void handleTap() {
  // Need to pop the navigator first in case onTap may push new route onto navigator.
  Navigator.pop<T>(context, widget.value);

  widget.onTap?.call();
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.27.4, on macOS 13.7.1 22H221 darwin-x64, locale en-NL)
    • Flutter version 3.27.4 on channel stable at /Users/yiou/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d8a9f9a52e (6 weeks ago), 2025-01-31 16:07:18 -0500
    • Engine revision 82bd5b7209
    • Dart version 3.6.2
    • DevTools version 2.40.3

[!] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at /Users/yiou/Library/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • ANDROID_HOME = /Users/yiou/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/yiou/Library/Android/sdk
    • Java binary at: /Users/yiou/Library/Java/JavaVirtualMachines/jbr-17.0.12/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment JBR-17.0.12+1-1207.37-nomod (build 17.0.12+1-b1207.37)
    ! Some Android licenses not accepted. To resolve this, run: flutter doctor --android-licenses

[!] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS and macOS development.
      Download at: https://developer.apple.com/xcode/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915915-b509.11)

[✓] VS Code (version 1.96.4)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.106.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.7.1 22H221 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 134.0.6998.89

[✓] Network resources
    • All expected network resources are available.```

</details>

**Comment by darshankawar** [comment]
@SexyHenk 
You are using third party implementation cubit. Please provide us code without such implementation (ex: main.dart) which still shows the reported behavior.

**Comment by SexyHenk** [comment]
Probably not the best example. When talkback/screenreader is active, i'll go to the previous page, if I try to tap the button. The focus is larger then the button, so maybe I have to exclude some things in semantics. This works without talkback/screenreader

`import 'package:flutter/material.dart';
import 'package:flutter_application_stories/centerWrap.dart';
import 'package:flutter_application_stories/theme_dimensions.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        title: 'Flutter Demo',
        debugShowCheckedModeBanner: false,
        theme: ThemeData(
          primarySwatch: Colors.green,
        ),
        home: const Home());
  }
}

class Home extends StatelessWidget {
  const Home({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: const Color(0xFF237F4C),
        title: const Text('Example'),
      ),
      body: SafeArea(
        child: SingleChildScrollView(
          child: CenterWrap(
            child: Padding(
              padding: ThemeDimensions.paddingAll16,
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: <Widget>[
                  OutlinedButton(
                    onPressed: () {
                      print('Button pressed'); // Debugging statement
                      Navigator.push(
                        context,
                        MaterialPageRoute(
                            builder: (context) => const SecondPage()),
                      );
                    },
                    child: const Text('Go to Second Page'),
                  ),
                ],
              ),
            ),
          ),
        ),
      ),
    );
  }
}

class SecondPage extends StatelessWidget {
  const SecondPage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Second Page'),
      ),
      body: Column(
        children: <Widget>[
          Padding(
            padding: ThemeDimensions.paddingAll16,
            child: PopupMenuItem<dynamic>(
              child: Semantics(
                child: TextButton(
                  onPressed: () {
                    print('Button tapped');
                  },
                  child: const Text('Tappable Button'),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}
`

**Comment by darshankawar** [comment]
> child: PopupMenuItem<dynamic>(

@SexyHenk You seem to be missing the `value` parameter ?  Check https://api.flutter.dev/flutter/material/PopupMenuItem-class.html for correct implementation.

**Comment by SexyHenk** [comment]
@darshankawar 

I added a value, but it didn't change the behaviour. It seems to be optional, if there's no need to return anything.

**Comment by darshankawar** [comment]
Thanks for the update. This is strange behavior that I was able to replicate with accessibility ON. Below is runnable code sample:

<details>
<summary> code </summary>

```

import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        title: 'Flutter Demo',
        debugShowCheckedModeBanner: false,
        theme: ThemeData(
          primarySwatch: Colors.green,
        ),
        home: const Home());
  }
}

class Home extends StatelessWidget {
  const Home({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: const Color(0xFF237F4C),
        title: const Text('Example'),
      ),
      body: SafeArea(
        child: SingleChildScrollView(
          child: Center(
            child: Padding(
              padding: EdgeInsets.all(16),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: <Widget>[
                  OutlinedButton(
                    onPressed: () {
                      print('Button pressed'); // Debugging statement
                      Navigator.push(
                        context,
                        MaterialPageRoute(
                            builder: (context) => const SecondPage()),
                      );
                    },
                    child: const Text('Go to Second Page'),
                  ),
                ],
              ),
            ),
          ),
        ),
      ),
    );
  }
}

class SecondPage extends StatelessWidget {
  const SecondPage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Second Page'),
      ),
      body: Column(
        children: <Widget>[
          Padding(
            padding: EdgeInsets.all(16),
            child: PopupMenuItem<dynamic>(
              child: Semantics(
                child: SizedBox(
                  width: double.infinity,
                  child: TextButton(
                    onPressed: () {
                      print('Button tapped');
                    },
                    child: const Text('Tappable Button'),
                  ),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}

```

</details>

```
stable : 3.29.2
master : 3.30.0-1.0.pre.526
```

**Comment by chunhtai** [comment]
My guess is the modal barrier received the click, but need to investigate more to find out
